packages/rum-recorder/src/domain/rrweb/record.ts

```diff
@@ -86,28 +80,28 @@ function record<T = EventWithTime>(options: RecordOptions<T> = {}): RecordAPI |
 
   polyfill()
 
-  let lastFullSnapshotEvent: EventWithTime
+  let lastFullSnapshotRecordTimestamp: number
   let incrementalSnapshotCount = 0
-  wrappedEmit = (e: EventWithTime, isCheckout?: boolean) => {
+  wrappedEmit = (record, isCheckout) => {
     if (
       mutationBuffer.isFrozen() &&
-      e.type !== EventType.FullSnapshot &&
-      !(e.type === EventType.IncrementalSnapshot && e.data.source === IncrementalSource.Mutation)
+      record.type !== RecordType.FullSnapshot &&
+      !(record.type === RecordType.IncrementalSnapshot && record.data.source === IncrementalSource.Mutation)
     ) {
-      // we've got a user initiated event so first we need to apply
+      // we've got a user initiated record so first we need to apply
       // all DOM changes that have been buffering during paused state
       mutationBuffer.emit()
       mutationBuffer.unfreeze()
     }
 
-    emit(((packFn ? packFn(e) : e) as unknown) as T, isCheckout)
-    if (e.type === EventType.FullSnapshot) {
-      lastFullSnapshotEvent = e
+    emit(((packFn ? packFn(record) : record) as unknown) as T, isCheckout)
+    if (record.type === RecordType.FullSnapshot) {
+      lastFullSnapshotRecordTimestamp = Date.now()
       incrementalSnapshotCount = 0
-    } else if (e.type === EventType.IncrementalSnapshot) {
+    } else if (record.type === RecordType.IncrementalSnapshot) {
       incrementalSnapshotCount += 1
       const exceedCount = checkoutEveryNth && incrementalSnapshotCount >= checkoutEveryNth
-      const exceedTime = checkoutEveryNms && e.timestamp - lastFullSnapshotEvent.timestamp > checkoutEveryNms
+      const exceedTime = checkoutEveryNms && Date.now() - lastFullSnapshotRecordTimestamp > checkoutEveryNms
```

Comment:
Why did you changed from `e.timestamp`to `Date.now()`?

Reply:
Because `rrweb/record` uses `RawRecords` instead of `Records` now, so records don't have the `timestamp` property anymore. They are "wraped" later in the recorder code. See the whole commit for context https://github.com/DataDog/browser-sdk/pull/707/commits/995f8635a49ba731e9c22e371a2f633225851071